blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpEventsIngestionHandler.java

```diff
@@ -65,40 +68,48 @@ public void handle(ChannelHandlerContext ctx, FullHttpRequest request) {
                     CharsetUtil.UTF_8);
             Event event = objectMapper.readValue(body, Event.class);
 
-            long timestamp = event.getWhen();
-            long current = System.currentTimeMillis();
-
-            if (event.getWhat().equals("")) {
-                throw new InvalidDataException(String.format( HttpMetricsIngestionHandler.ERROR_HEADER + System.lineSeparator() + "Event should contain at least '%s' field.", Event.FieldLabels.what.name()));
+            // To verify if the request has multiple elements. If some one sends request with multiple root elements,
+            // parser does not throw any validation exceptions. We are checking it manually here.
+            Iterator<Event> iterator = objectMapper.reader(Event.class).readValues(body);
+            if (iterator.hasNext()) {
+                iterator.next();
+                if (iterator.hasNext()) { //has more than one element
+                    throw new InvalidDataException("Only one event is allowed per request");
+                }
             }
 
-            if ( timestamp > current + futureDiff ) {
-                throw new InvalidDataException( HttpMetricsIngestionHandler.ERROR_HEADER + System.lineSeparator() + "'" + event.getWhat() + "': 'when' '" + timestamp + "' is more than '" + futureDiff + "' milliseconds into the future." );
+            Set<ConstraintViolation<Event>> constraintViolations = validator.validate(event);
+
+            List<ErrorResponse.ErrorData> validationErrors = new ArrayList<ErrorResponse.ErrorData>();
+            for (ConstraintViolation<Event> constraintViolation : constraintViolations) {
+                validationErrors.add(new ErrorResponse.ErrorData(tenantId, "",
+                        constraintViolation.getPropertyPath().toString(), constraintViolation.getMessage()));
             }
 
-            if ( timestamp < current - pastDiff ) {
-                throw new InvalidDataException( HttpMetricsIngestionHandler.ERROR_HEADER + System.lineSeparator() + "'" + event.getWhat() + "': 'when' '" + timestamp + "' is more than '" + pastDiff + "' milliseconds into the past." );
+            if (!validationErrors.isEmpty()) {
+                DefaultHandler.sendErrorResponse(ctx, request, validationErrors, HttpResponseStatus.BAD_REQUEST);
+                return;
             }
 
             searchIO.insert(tenantId, Arrays.asList(event.toMap()));
+            DefaultHandler.sendResponse(ctx, request, response, HttpResponseStatus.OK);
         } catch (JsonMappingException e) {
             log.error(String.format("Exception %s", e.toString()));
```

Comment:
do we want to log BAD_REQUESTs as errors?  Do we need to log these at all?  We'll see the status code in our access logs.